infra/aws/govcloud.md

```diff
@@ -0,0 +1,199 @@
+* Commercial AWS and GovCloud resources cannot interact directly in any
+  way. This means that you cannot assume roles between the two or tie
+  a commercial resource to a GovCloud resource unless using some publicly
+  addressable method.
+* GovCloud resources use a different AWS partition; this is the second
+  part of an ARN. So while an ARN for a commercial resource might look
+  like this:
+
+  ```text
+  arn:aws:iam::123456789000:user/testuser
+  ```
+
+  a GovCloud ARN will look like this:
+
+  ```text
+  arn:aws-us-gov:iam::123456789000:user/testuser
+  ```
+
+  This means that any Terraform modules will need to take this into
+  account; see the Terraform
+  [`aws_partition`](https://www.terraform.io/docs/providers/aws/d/partition.html)
+  data source for how to do this.
+* Many AWS services are not available in GovCloud; some are available
+  in one GovCloud region and not others. See the GovCloud
+  [Services in Scope](https://aws.amazon.com/compliance/services-in-scope/)
+  and the AWS [Region
+  Table](https://aws.amazon.com/about-aws/global-infrastructure/regional-product-services/)
+  for more information.
+* Cloudtrail is automatically added to all accounts in GovCloud by default.
+  This means that you will have to handle things differently when trying to
+  configure an organization's Cloudtrail.
+
+## Setting Up a GovCloud Organization
+
+### Create Your GovCloud Organization Master Account
+
+To set up a new GovCloud AWS Organization, you must log in as root to the
+master account of your commercial AWS Organization. Then go to the "My
+Account" screen with the pulldown in the upper right of the AWS dashboard,
+and scroll down; there will be a "Sign up for AWS GovCloud" button. When
+you click that, you will need to accept an appendix to the AWS Customer
+Agreement; once you do so, you will get an email with instructions on how
+to log in to your GovCloud account.
+
+Once you create the account, you can create a temporary IAM user with full
+credentials you can use to bootstrap Terraform in the account as you would
+in commercial AWS.
+
+Once you have bootstrapped Terraform, you can create the GovCloud
+organization the same way you would in commercial; however, you cannot
+add `cloudtrail.amazonaws.com` as a service access principal in your
+organization definition due to the way Cloudtrail is automatically added
+to all GovCloud accounts.
+
+### Creating Additional GovCloud Accounts
+
+To add more accounts to your GovCloud organization, you *cannot* simply
+use the Terraform `aws_organizations_account` resource. Using credentials
+for the master account in your commercial AWS Organization, use the
+following AWS CLI command. This will create two accounts; a dummy account
+in the commercial AWS Organization and a new corresponding GovCloud
+account, both with the same account alias.
+
+```console
+$ aws organizations create-gov-cloud-account \
+  --email spacecats-infra+spacecats-govcloud-id@example.com \
```

Comment:
fucking spacecats

Reply:
is this just any email from `truss.works`? And is the account name just the AWS Account ID?

Reply:
@chrisgilmerproj The email is the email address you want to have AWS send notifications about the account to. So in this case, `spacecats-infra@example.com` is the email alias for the infra team on the Spacecats project (with a suffix for that particular GovCloud account), and `spacecats-govcloud-id` is just the account alias for the new account.

Reply:
Aha! I'd put a note about that somewhere here. Also maybe point out that using the `+` is a good way to identify senders.